src/main/java/teammates/ui/controller/InstructorCourseJoinConfirmationPageData.java

```diff
@@ -1,14 +1,43 @@
 package teammates.ui.controller;
 
 import teammates.common.datatransfer.AccountAttributes;
+import teammates.common.util.Const;
+import teammates.common.util.Sanitizer;
 
 public class InstructorCourseJoinConfirmationPageData extends PageData {
 
-    public String regkey;
-    public String institute;
+    private String regkey;
+    private String institute;
     
     public InstructorCourseJoinConfirmationPageData(AccountAttributes account) {
         super(account);
     }
+    
+    public void init(String regkey, String institute) {
+        this.regkey = regkey;
+        this.institute = institute;
+    }
+    
+    public String getRegkey() {
+        return regkey;
+    }
+    
+    public String getInstitue() {
+        return institute;
+    }
+    
+    public String getConfirmationLink() {
+        String ref = "";
+        if (institute == null) {
 
+            ref = Const.ActionURIs.INSTRUCTOR_COURSE_JOIN_AUTHENTICATED
+                    + "?key=" + regkey;
+        } else {
+            ref = Const.ActionURIs.INSTRUCTOR_COURSE_JOIN_AUTHENTICATED
+                    + "?key=" + regkey + "&"
```

Comment:
Would it be better to set it to
`Const.ActionURIs.INSTRUCTOR_COURSE_JOIN_AUTHENTICATED + "?key=" + regkey`
first and then append the rest if `institute != null`?